### PR TITLE
#866aa0wkr add endpoint for removing package

### PR DESCRIPTION
**Pull request by bilaldbank**

### CODE REVIEW

File: unpub/lib/src/app.dart [hunk 1]
```diff
@@ -510,10 +532,19 @@ class App {
...
   Future<shelf.Response> indexHtml(shelf.Request req) async {
-    return shelf.Response.ok(index_html.content,
+    final updatedHtmlContent = _addStartTimeInHtml();
+
+    return shelf.Response.ok(updatedHtmlContent,
         headers: {HttpHeaders.contentTypeHeader: ContentType.html.mimeType});
   }
 
+  /// Update the title in HTML to include server start time
+  String _addStartTimeInHtml() {
```
Comment (bilaldbank) [inline unpub/lib/src/app.dart:549]: This just adds the server app start time in title so we can be sure when it gets updated

File: unpub/lib/src/app.dart [hunk 2]
```diff
@@ -379,6 +382,25 @@ class App {
     return _successMessage('uploader removed');
   }
 
+  @Route.delete('/api/packages/<name>')
+  Future<shelf.Response> removePackage(shelf.Request req, String name) async {
+    try {
+      await _validateOpaqueToken(req);
```
Comment (bilaldbank) [inline unpub/lib/src/app.dart:388]: Validate opaque token when a delete request is received

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,18 @@
+FROM google/dart:2.12 AS builder
```
Comment (bilaldbank) [inline Dockerfile]: @tasnim-din We need to use Dart 2.12 to build the project

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,18 @@
...
+
+WORKDIR /app
+COPY unpub_web/pubspec.* ./
+RUN dart pub get
+
+COPY . .
```
Comment (bilaldbank) [inline Dockerfile]: Copying the parent directory because unpub_web uses unpub_api

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,18 @@
...
+
+WORKDIR /app/unpub_web
+
+RUN dart pub get --offline
+
+RUN dart pub global activate webdev 2.7.4
+
+RUN dart pub global run webdev build
```
Comment (bilaldbank) [inline Dockerfile]: This should create a build folder we can use in unpub as static web files

File: Dockerfile [hunk 6]
```diff
@@ -0,0 +1,18 @@
+FROM google/dart:2.12 AS builder
+
+WORKDIR /app
+COPY unpub_web/pubspec.* ./
+RUN dart pub get
```
Comment (bilaldbank) [inline Dockerfile]: Currently getting exception on this

**Comment by bilaldbank** [comment]
Will add a separate PR for unpub_web changes to support a delete button in the UI. The endpoint can be reviewed and tested separately

**Review by tasnim-din — APPROVED** [review]

**Pull request merged**